packages/core/src/operations.ts

```diff
@@ -103,7 +103,7 @@ export class TxOperations implements Omit<Client, 'notify'> {
     space: Ref<Space>,
     attachedTo: Ref<T>,
     attachedToClass: Ref<Class<T>>,
-    collection: string,
+    collection: Extract<keyof T, string> | string,
```

Comment:
Strictly speaking it should be just `Extract<keyof T, string>`, without `| string`, but I could not fix it quickly because it affected too many places, and brought too many warnings/errors. So for now it only helps IDE autocomplete better, but doesn't help TS to do type check